### PR TITLE
java well know problem -Palindrome number and string.

### PR DESCRIPTION
**Pull request by PradeepFSTdhane123**
### Description
palindrome problem string and number added.

Fixes # 805

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
Please read our contributions guideline carefully. We have provided all of our resources, such as `.md` or `.mdx` syntax, our documentation format, and much more.

**Comment by PradeepFSTdhane123** [comment]
@ajay-dhangar I am also created fine in  .md  format , whart is the actual reason can you tell? so That I will improve it

**Comment by PradeepFSTdhane123** [comment]
my file format is .md so what is the issue for not mearching

File: java/Palindrome_number.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+## Description
+Given a number , check is it palindrome or not , and return answer in boolean. Palindrome is when we read it from left to right and right to left it same. ex-121.
+
+## Approach
+
+(rev = rev * 10 + x % 10; and x = x / 10;)
+//Modulo operator (%) is used to extract the last digit of the number.
+// division (/) is used to remove the last digit of the number.
+//Palindrome property: A number is a palindrome if it reads the same forward and backward.
+
```
Comment (ajay-dhangar) [inline java/Palindrome_number.md]: can you write this in good way

File: java/Palindrome_number.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        int original=num;  //redefined the value of num.
+        while(num>0){       //while loop upto the num greater than 0
+          rev=rev*10+num%10;   //this is logic for reverse the number.
+          num=num/10;  //last digit deleted.
+        } 
+        return rev==original;  // check 
+      }
+
+}
+
```
Comment (ajay-dhangar) [inline java/Palindrome_number.md]: Write your code example like 

````md
```java
// your code here
```
````

File: java/Palindrome_string.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      if(str.charAt(left)!=str.charAt(right)){
+        return false;
+      }
+      left++;      //left increment 
+      right--;     //right decrement.
+    }
+    return true;   //i.e all character mathes.
+  }
+}
+
```
Comment (ajay-dhangar) [inline java/Palindrome_string.md]: same mistake

File: java/Palindrome_string.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+**👉🏻👉🏻👉🏻 time complexity**= O(n);
+**👉🏻👉🏻👉🏻space complexity**= O(1);
+**👉🏻👉🏻👉🏻logic =**
+//1.The key logic is the use of two pointers (left and right) to compare characters from both ends of the string and progressively move toward the center.
+//2.If all characters match, the string is a palindrome. If any pair of characters does not match, the string is not a palindrome.
+
```
Comment (ajay-dhangar) [inline java/Palindrome_string.md]: text is not looking good  why you use `//`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
change your code based on suggestions

**Comment by PradeepFSTdhane123** [comment]
@ajay-dhangar  solve the conflict kindely merge it

**Pull request closed without merging**